bigquery/client.py

```diff
@@ -120,17 +120,38 @@ def __init__(self, bq_service, project_id, swallow_results=True):
         self.swallow_results = swallow_results
         self.cache = {}
 
-    def query(self, query, max_results=None, timeout=0, dry_run=False):
-        """Submit a query to BigQuery.
+    def _submit_job(self, body_object):
+
+        """ Submit a query job to BigQuery
+
+            This is similar to BigQueryClient.query, but gives the user
 
         Args:
-            query: BigQuery query string.
-            max_results: maximum number of rows to return per page of results.
-            timeout: how long to wait for the query to complete, in seconds,
-                     before the request times out and returns.
-            dry_run: if True, the query isn't actually run. A valid query will
-                     return an empty response, while an invalid one will return
-                     the same error message it would if it wasn't a dry run.
+            body_object: body object as per these docs
+                        https://google-api-client-libraries.appspot.com/documentation/bigquery/v2/python/latest/bigquery_v2.jobs.html#insert
+
+        Returns:
+            response of the bigquery.jobs().insert().execute() call
+
+        Raises:
+            BigQueryTimeoutException on timeout
+        """
+
+        logging.debug('Submitting job: %s' % body_object)
+
+        job_collection = self.bigquery.jobs()
+
+        return job_collection.insert(projectId=self.project_id, body=body_object).execute()
+
+    def _submit_query_job(self, query_data):
+
+        """ Submit a query job to BigQuery
+
+            This is similar to BigQueryClient.query, but gives the user
```

Comment:
Same note as above. Also might want to document the difference between `_submit_query_job` and `_submit_job` since they look similar.